src/scripts.js

```diff
@@ -1,12 +1,129 @@
-var assert = require('assert')
-var ops = require('./opcodes')
-var typeForce = require('typeforce')
+var bufferutils = require('./bufferutils')
+var typeforce = require('typeforce')
+var types = require('./types')
 
+var ECSignature = require('./ecsignature')
 var ecurve = require('ecurve')
 var curve = ecurve.getCurveByName('secp256k1')
 
-var ECSignature = require('./ecsignature')
-var Script = require('./script')
+var OPS = require('./opcodes')
+var REVERSE_OPS = []
+for (var op in OPS) {
+  var code = OPS[op]
+  REVERSE_OPS[code] = op
+}
+
+function toASM (chunks) {
+  if (types.Buffer(chunks)) {
+    chunks = decompile(chunks)
+  }
+
+  return chunks.map(function (chunk) {
+    // data chunk
+    if (Buffer.isBuffer(chunk)) {
+      return chunk.toString('hex')
+
+    // opcode
+    } else {
+      return REVERSE_OPS[chunk]
+    }
+  }).join(' ')
+}
+
+function fromASM (asm) {
+  typeforce(types.String, asm)
+
+  var strChunks = asm.split(' ')
+  var chunks = strChunks.map(function (strChunk) {
+    // opcode
+    if (strChunk in OPS) {
+      return OPS[strChunk]
+
+    // data chunk
+    } else {
+      return new Buffer(strChunk, 'hex')
+    }
+  })
+
+  return compile(chunks)
+}
+
+function compile (chunks) {
```

Comment:
From my point of view, it's not obvious what are the input and output types just by reading the function name.